tests/testsuite/tree.rs

```diff
@@ -1085,6 +1085,57 @@ fn duplicates_with_target() {
     p.cargo("tree -d --target=all").with_stdout("").run();
 }
 
+#[cargo_test]
+fn duplicates_with_proc_macro() {
+    Package::new("cat", "1.0.0").publish();
+    Package::new("cat", "2.0.0").publish();
+    Package::new("dep", "1.0.0")
+        .proc_macro(true)
+        .dep("cat", "1.0")
+        .publish();
+    let p = project()
```

Comment:
nit: could you rename these to be based on their role within the test so its easier to scan?
- `dep` -> `proc`
- `cat@1.0.0` -> `proc-dep`
- `cat@2.0.0` -> `normal;-dep`

EDIT: Ah, `cat` is intended to specifically be a duplicate.

Maybe we should instead have those **and** `proc-dep` and `normal-dep` and name them `dupe-dep`

Reply:
This is based on one of the previous test cases, but I agree that more explicit names would help. I don't think `normal-dep` and `proc-dep` add anything for a find-duplicates test, though.